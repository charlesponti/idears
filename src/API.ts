/* tslint:disable */
/* eslint-disable */
//  This file was automatically generated and should not be edited.

export type CreateIdeaInput = {
  id?: string | null,
  title: string,
  description?: string | null,
  status?: string | null,
  upvotes?: number | null,
  downvotes?: number | null,
  userID: string,
};

export type ModelIdeaConditionInput = {
  title?: ModelStringInput | null,
  description?: ModelStringInput | null,
  status?: ModelStringInput | null,
  upvotes?: ModelIntInput | null,
  downvotes?: ModelIntInput | null,
  userID?: ModelIDInput | null,
  and?: Array< ModelIdeaConditionInput | null > | null,
  or?: Array< ModelIdeaConditionInput | null > | null,
  not?: ModelIdeaConditionInput | null,
};

export type ModelStringInput = {
  ne?: string | null,
  eq?: string | null,
  le?: string | null,
  lt?: string | null,
  ge?: string | null,
  gt?: string | null,
  contains?: string | null,
  notContains?: string | null,
  between?: Array< string | null > | null,
  beginsWith?: string | null,
  attributeExists?: boolean | null,
  attributeType?: ModelAttributeTypes | null,
  size?: ModelSizeInput | null,
};

export enum ModelAttributeTypes {
  binary = "binary",
  binarySet = "binarySet",
  bool = "bool",
  list = "list",
  map = "map",
  number = "number",
  numberSet = "numberSet",
  string = "string",
  stringSet = "stringSet",
  _null = "_null",
}


export type ModelSizeInput = {
  ne?: number | null,
  eq?: number | null,
  le?: number | null,
  lt?: number | null,
  ge?: number | null,
  gt?: number | null,
  between?: Array< number | null > | null,
};

export type ModelIntInput = {
  ne?: number | null,
  eq?: number | null,
  le?: number | null,
  lt?: number | null,
  ge?: number | null,
  gt?: number | null,
  between?: Array< number | null > | null,
  attributeExists?: boolean | null,
  attributeType?: ModelAttributeTypes | null,
};

export type ModelIDInput = {
  ne?: string | null,
  eq?: string | null,
  le?: string | null,
  lt?: string | null,
  ge?: string | null,
  gt?: string | null,
  contains?: string | null,
  notContains?: string | null,
  between?: Array< string | null > | null,
  beginsWith?: string | null,
  attributeExists?: boolean | null,
  attributeType?: ModelAttributeTypes | null,
  size?: ModelSizeInput | null,
};

export type UpdateIdeaInput = {
  id: string,
  title?: string | null,
  description?: string | null,
  status?: string | null,
  upvotes?: number | null,
  downvotes?: number | null,
  userID?: string | null,
};

export type DeleteIdeaInput = {
  id?: string | null,
};

export type CreateUserInput = {
  id?: string | null,
  email: string,
};

export type ModelUserConditionInput = {
  email?: ModelStringInput | null,
  and?: Array< ModelUserConditionInput | null > | null,
  or?: Array< ModelUserConditionInput | null > | null,
  not?: ModelUserConditionInput | null,
};

export type UpdateUserInput = {
  id: string,
  email?: string | null,
};

export type DeleteUserInput = {
  id?: string | null,
};

export type CreateIdeaVoteInput = {
  id?: string | null,
  ideaID: string,
  userID: string,
  vote?: number | null,
};

export type ModelIdeaVoteConditionInput = {
  ideaID?: ModelIDInput | null,
  userID?: ModelIDInput | null,
  vote?: ModelIntInput | null,
  and?: Array< ModelIdeaVoteConditionInput | null > | null,
  or?: Array< ModelIdeaVoteConditionInput | null > | null,
  not?: ModelIdeaVoteConditionInput | null,
};

export type UpdateIdeaVoteInput = {
  id: string,
  ideaID?: string | null,
  userID?: string | null,
  vote?: number | null,
};

export type DeleteIdeaVoteInput = {
  id?: string | null,
};

export type ModelIdeaFilterInput = {
  id?: ModelIDInput | null,
  title?: ModelStringInput | null,
  description?: ModelStringInput | null,
  status?: ModelStringInput | null,
  upvotes?: ModelIntInput | null,
  downvotes?: ModelIntInput | null,
  userID?: ModelIDInput | null,
  and?: Array< ModelIdeaFilterInput | null > | null,
  or?: Array< ModelIdeaFilterInput | null > | null,
  not?: ModelIdeaFilterInput | null,
};

export type ModelUserFilterInput = {
  id?: ModelIDInput | null,
  email?: ModelStringInput | null,
  and?: Array< ModelUserFilterInput | null > | null,
  or?: Array< ModelUserFilterInput | null > | null,
  not?: ModelUserFilterInput | null,
};

export type ModelIdeaVoteFilterInput = {
  id?: ModelIDInput | null,
  ideaID?: ModelIDInput | null,
  userID?: ModelIDInput | null,
  vote?: ModelIntInput | null,
  and?: Array< ModelIdeaVoteFilterInput | null > | null,
  or?: Array< ModelIdeaVoteFilterInput | null > | null,
  not?: ModelIdeaVoteFilterInput | null,
};

export type CreateIdeaMutationVariables = {
  input: CreateIdeaInput,
  condition?: ModelIdeaConditionInput | null,
};

export type CreateIdeaMutation = {
  createIdea:  {
    __typename: "Idea",
    id: string,
    title: string,
    description: string | null,
    status: string | null,
    upvotes: number | null,
    downvotes: number | null,
    userID: string,
    user:  {
      __typename: "User",
      id: string,
      email: string,
      createdAt: string,
      updatedAt: string,
      owner: string | null,
    } | null,
    createdAt: string,
    updatedAt: string,
    owner: string | null,
  } | null,
};

export type UpdateIdeaMutationVariables = {
  input: UpdateIdeaInput,
  condition?: ModelIdeaConditionInput | null,
};

export type UpdateIdeaMutation = {
  updateIdea:  {
    __typename: "Idea",
    id: string,
    title: string,
    description: string | null,
    status: string | null,
    upvotes: number | null,
    downvotes: number | null,
    userID: string,
    user:  {
      __typename: "User",
      id: string,
      email: string,
      createdAt: string,
      updatedAt: string,
      owner: string | null,
    } | null,
    createdAt: string,
    updatedAt: string,
    owner: string | null,
  } | null,
};

export type DeleteIdeaMutationVariables = {
  input: DeleteIdeaInput,
  condition?: ModelIdeaConditionInput | null,
};

export type DeleteIdeaMutation = {
  deleteIdea:  {
    __typename: "Idea",
    id: string,
    title: string,
    description: string | null,
    status: string | null,
    upvotes: number | null,
    downvotes: number | null,
    userID: string,
    user:  {
      __typename: "User",
      id: string,
      email: string,
      createdAt: string,
      updatedAt: string,
      owner: string | null,
    } | null,
    createdAt: string,
    updatedAt: string,
    owner: string | null,
  } | null,
};

export type CreateUserMutationVariables = {
  input: CreateUserInput,
  condition?: ModelUserConditionInput | null,
};

export type CreateUserMutation = {
  createUser:  {
    __typename: "User",
    id: string,
    email: string,
    ideas:  {
      __typename: "ModelIdeaConnection",
      nextToken: string | null,
    } | null,
    createdAt: string,
    updatedAt: string,
    owner: string | null,
  } | null,
};

export type UpdateUserMutationVariables = {
  input: UpdateUserInput,
  condition?: ModelUserConditionInput | null,
};

export type UpdateUserMutation = {
  updateUser:  {
    __typename: "User",
    id: string,
    email: string,
    ideas:  {
      __typename: "ModelIdeaConnection",
      nextToken: string | null,
    } | null,
    createdAt: string,
    updatedAt: string,
    owner: string | null,
  } | null,
};

export type DeleteUserMutationVariables = {
  input: DeleteUserInput,
  condition?: ModelUserConditionInput | null,
};

export type DeleteUserMutation = {
  deleteUser:  {
    __typename: "User",
    id: string,
    email: string,
    ideas:  {
      __typename: "ModelIdeaConnection",
      nextToken: string | null,
    } | null,
    createdAt: string,
    updatedAt: string,
    owner: string | null,
  } | null,
};

export type CreateIdeaVoteMutationVariables = {
  input: CreateIdeaVoteInput,
  condition?: ModelIdeaVoteConditionInput | null,
};

export type CreateIdeaVoteMutation = {
  createIdeaVote:  {
    __typename: "IdeaVote",
    id: string,
    ideaID: string,
    userID: string,
    vote: number | null,
    createdAt: string,
    updatedAt: string,
    owner: string | null,
  } | null,
};

export type UpdateIdeaVoteMutationVariables = {
  input: UpdateIdeaVoteInput,
  condition?: ModelIdeaVoteConditionInput | null,
};

export type UpdateIdeaVoteMutation = {
  updateIdeaVote:  {
    __typename: "IdeaVote",
    id: string,
    ideaID: string,
    userID: string,
    vote: number | null,
    createdAt: string,
    updatedAt: string,
    owner: string | null,
  } | null,
};

export type DeleteIdeaVoteMutationVariables = {
  input: DeleteIdeaVoteInput,
  condition?: ModelIdeaVoteConditionInput | null,
};

export type DeleteIdeaVoteMutation = {
  deleteIdeaVote:  {
    __typename: "IdeaVote",
    id: string,
    ideaID: string,
    userID: string,
    vote: number | null,
    createdAt: string,
    updatedAt: string,
    owner: string | null,
  } | null,
};

export type GetIdeaQueryVariables = {
  id: string,
};

export type GetIdeaQuery = {
  getIdea:  {
    __typename: "Idea",
    id: string,
    title: string,
    description: string | null,
    status: string | null,
    upvotes: number | null,
    downvotes: number | null,
    userID: string,
    user:  {
      __typename: "User",
      id: string,
      email: string,
      createdAt: string,
      updatedAt: string,
      owner: string | null,
    } | null,
    createdAt: string,
    updatedAt: string,
    owner: string | null,
  } | null,
};

export type ListIdeasQueryVariables = {
  filter?: ModelIdeaFilterInput | null,
  limit?: number | null,
  nextToken?: string | null,
};

export type ListIdeasQuery = {
  listIdeas:  {
    __typename: "ModelIdeaConnection",
    items:  Array< {
      __typename: "Idea",
      id: string,
      title: string,
      description: string | null,
      status: string | null,
      upvotes: number | null,
      downvotes: number | null,
      userID: string,
      createdAt: string,
      updatedAt: string,
      owner: string | null,
    } | null > | null,
    nextToken: string | null,
  } | null,
};

export type GetUserQueryVariables = {
  id: string,
};

export type GetUserQuery = {
  getUser:  {
    __typename: "User",
    id: string,
    email: string,
    ideas:  {
      __typename: "ModelIdeaConnection",
      nextToken: string | null,
    } | null,
    createdAt: string,
    updatedAt: string,
    owner: string | null,
  } | null,
};

export type ListUsersQueryVariables = {
  filter?: ModelUserFilterInput | null,
  limit?: number | null,
  nextToken?: string | null,
};

export type ListUsersQuery = {
  listUsers:  {
    __typename: "ModelUserConnection",
    items:  Array< {
      __typename: "User",
      id: string,
      email: string,
      createdAt: string,
      updatedAt: string,
      owner: string | null,
    } | null > | null,
    nextToken: string | null,
  } | null,
};

export type GetIdeaVoteQueryVariables = {
  id: string,
};

export type GetIdeaVoteQuery = {
  getIdeaVote:  {
    __typename: "IdeaVote",
    id: string,
    ideaID: string,
    userID: string,
    vote: number | null,
    createdAt: string,
    updatedAt: string,
    owner: string | null,
  } | null,
};

export type ListIdeaVotesQueryVariables = {
  filter?: ModelIdeaVoteFilterInput | null,
  limit?: number | null,
  nextToken?: string | null,
};

export type ListIdeaVotesQuery = {
  listIdeaVotes:  {
    __typename: "ModelIdeaVoteConnection",
    items:  Array< {
      __typename: "IdeaVote",
      id: string,
      ideaID: string,
      userID: string,
      vote: number | null,
      createdAt: string,
      updatedAt: string,
      owner: string | null,
    } | null > | null,
    nextToken: string | null,
  } | null,
};

export type OnCreateIdeaSubscriptionVariables = {
  owner: string,
};

export type OnCreateIdeaSubscription = {
  onCreateIdea:  {
    __typename: "Idea",
    id: string,
    title: string,
    description: string | null,
    status: string | null,
    upvotes: number | null,
    downvotes: number | null,
    userID: string,
    user:  {
      __typename: "User",
      id: string,
      email: string,
      createdAt: string,
      updatedAt: string,
      owner: string | null,
    } | null,
    createdAt: string,
    updatedAt: string,
    owner: string | null,
  } | null,
};

export type OnUpdateIdeaSubscriptionVariables = {
  owner: string,
};

export type OnUpdateIdeaSubscription = {
  onUpdateIdea:  {
    __typename: "Idea",
    id: string,
    title: string,
    description: string | null,
    status: string | null,
    upvotes: number | null,
    downvotes: number | null,
    userID: string,
    user:  {
      __typename: "User",
      id: string,
      email: string,
      createdAt: string,
      updatedAt: string,
      owner: string | null,
    } | null,
    createdAt: string,
    updatedAt: string,
    owner: string | null,
  } | null,
};

export type OnDeleteIdeaSubscriptionVariables = {
  owner: string,
};

export type OnDeleteIdeaSubscription = {
  onDeleteIdea:  {
    __typename: "Idea",
    id: string,
    title: string,
    description: string | null,
    status: string | null,
    upvotes: number | null,
    downvotes: number | null,
    userID: string,
    user:  {
      __typename: "User",
      id: string,
      email: string,
      createdAt: string,
      updatedAt: string,
      owner: string | null,
    } | null,
    createdAt: string,
    updatedAt: string,
    owner: string | null,
  } | null,
};

export type OnCreateUserSubscriptionVariables = {
  owner: string,
};

export type OnCreateUserSubscription = {
  onCreateUser:  {
    __typename: "User",
    id: string,
    email: string,
    ideas:  {
      __typename: "ModelIdeaConnection",
      nextToken: string | null,
    } | null,
    createdAt: string,
    updatedAt: string,
    owner: string | null,
  } | null,
};

export type OnUpdateUserSubscriptionVariables = {
  owner: string,
};

export type OnUpdateUserSubscription = {
  onUpdateUser:  {
    __typename: "User",
    id: string,
    email: string,
    ideas:  {
      __typename: "ModelIdeaConnection",
      nextToken: string | null,
    } | null,
    createdAt: string,
    updatedAt: string,
    owner: string | null,
  } | null,
};

export type OnDeleteUserSubscriptionVariables = {
  owner: string,
};

export type OnDeleteUserSubscription = {
  onDeleteUser:  {
    __typename: "User",
    id: string,
    email: string,
    ideas:  {
      __typename: "ModelIdeaConnection",
      nextToken: string | null,
    } | null,
    createdAt: string,
    updatedAt: string,
    owner: string | null,
  } | null,
};

export type OnCreateIdeaVoteSubscriptionVariables = {
  owner: string,
};

export type OnCreateIdeaVoteSubscription = {
  onCreateIdeaVote:  {
    __typename: "IdeaVote",
    id: string,
    ideaID: string,
    userID: string,
    vote: number | null,
    createdAt: string,
    updatedAt: string,
    owner: string | null,
  } | null,
};

export type OnUpdateIdeaVoteSubscriptionVariables = {
  owner: string,
};

export type OnUpdateIdeaVoteSubscription = {
  onUpdateIdeaVote:  {
    __typename: "IdeaVote",
    id: string,
    ideaID: string,
    userID: string,
    vote: number | null,
    createdAt: string,
    updatedAt: string,
    owner: string | null,
  } | null,
};

export type OnDeleteIdeaVoteSubscriptionVariables = {
  owner: string,
};

export type OnDeleteIdeaVoteSubscription = {
  onDeleteIdeaVote:  {
    __typename: "IdeaVote",
    id: string,
    ideaID: string,
    userID: string,
    vote: number | null,
    createdAt: string,
    updatedAt: string,
    owner: string | null,
  } | null,
};
