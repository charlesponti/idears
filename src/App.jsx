import Amplify from 'aws-amplify'
import classnames from 'classnames'
import React from 'react'
import styles from './App.module.scss'
import awsExports from './aws-exports'
import brain from './brain.svg'
import Application from './containers/Application'
import UserProvider from './providers/UserProvider'

Amplify.configure(awsExports)

function App () {
  const Brain = () => (
    <img src={brain} className={styles.brain} alt="brain emoji"/>
  )

  return (
    <UserProvider>
      <div className={styles.container}>
        <header className={styles.header}>
          <Brain />
        </header>
        <main className={classnames(styles.main, 'mt-8')}>
          <Application/>
        </main>
      </div>
    </UserProvider>
  )
}

export default App
