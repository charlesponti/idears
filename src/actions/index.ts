export const auth = () => {

}

auth.sendPasswordResetEmail = (email: String) => {

}

auth.signInWithEmailAndPassword = (email: String, password: String) => {

}

auth.createUserWithEmailAndPassword = (email: String, password: String) => {
  return { user: { } }
}

auth.onAuthStateChanged = (fn: Function) => {
  fn()
}

export const getUser = () => {

}

export async function addIdeaToUser (id: string) {

}

export const addVoteToUser = async (ideaUID: string) => {

}

export async function addVoteToIdea (id: String, amount: Number) {

}

export async function saveIdea (idea: { title: String, description: String }) {

}

export const signInWithGoogle = () => {

}

export const signOut = () => {

}

export const getIdeas = async () => {

}

export const generateUserDocument = async (user: { }, additionalData?: any) => {

}
