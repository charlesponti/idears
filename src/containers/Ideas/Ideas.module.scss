@import "../../styles/colors.module.scss";

.container {
  padding-top: 2rem;
  border-top: 4px solid $grey;
  margin-top: 2rem;
}