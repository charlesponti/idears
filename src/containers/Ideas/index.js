import Button from '@material-ui/core/Button'
import Grid from '@material-ui/core/Grid'
import { navigate } from '@reach/router'
import React from 'react'
import IdeaListItem from 'src/components/IdeaListItem'
import useIdeas from 'src/hooks/useIdeas'
import { IdeasProvider } from 'src/providers/IdeasProvider'
import { appStates } from 'src/reducers/action-types'
import styles from './Ideas.module.scss'

export function Ideas () {
  const { ideas, status, error } = useIdeas()

  if (status === appStates.LOADING) return <div>Loading...</div>

  if (status === appStates.ERROR) return <div>Error... {error}</div>

  function onCreateIdeaClick () {
    navigate('ideas/create')
  }
  return (
    <Grid container justify="center" direction="column" className={styles.container}>
      <Grid item style={{ marginBottom: '1rem' }}>
        <Button color="primary" variant="contained" onClick={onCreateIdeaClick}>
          Post New Idea
        </Button>
      </Grid>
      <Grid item>
        {ideas.length ? ideas.map(idea => <IdeaListItem key={idea.id} idea={idea}/>) : null}
      </Grid>
    </Grid>
  )
}

function IdeasContainer () {
  return (
    <IdeasProvider>
      <Ideas/>
    </IdeasProvider>
  )
}

export default IdeasContainer
