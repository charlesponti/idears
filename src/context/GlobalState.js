import React, { createContext, useReducer } from 'react'
import t from 'prop-types'
import AppReducer from '../reducers/AppReducer'
import { getActions } from 'src/reducers/action-types'

// Initial State
const initialState = {
  user: {
    _id: 'SOME_ID',
    votes: [],
    ideas: []
  },
  ideas: [
    {
      _id: 0,
      title: 'Some Crazy Idea',
      description: 'An elaborate description of why this idea will change the world!',
      votes: 0
    }
  ]
}

// Context
export const GlobalContext = createContext(initialState)

// Provider
export const GlobalProvider = ({ children }) => {
  const [state, dispatch] = useReducer(AppReducer, initialState)

  return (
    <GlobalContext.Provider value={{ ...state, ...getActions(dispatch) }}>
      {children}
    </GlobalContext.Provider>
  )
}

GlobalProvider.propTypes = {
  children: t.node
}
