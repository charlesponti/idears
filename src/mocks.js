export function getMockState() {
  return {
    user: {
      _id: 'USER_ID'
    },
    ideas: [
      { 
        _id: 0, 
        title: 'Some idea', 
        description: 'Some description', 
        votes: 0 
      },
      { 
        _id: 1, 
        title: 'Some another idea', 
        description: 'Some another description', 
        votes: 5 
      }
    ]
  }
}