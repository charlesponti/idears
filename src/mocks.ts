import { IdeasState } from 'src/interfaces/Idea'

export interface MockUserState {
  uid: string;
  ideas: string[];
  votes: string[];
}

export function getMockUserState (): MockUserState {
  return {
    uid: 'USER_ID',
    ideas: [],
    votes: []
  }
}

export function getMockState (): IdeasState {
  return {
    isLoading: true,
    ideas: [
      {
        id: '0',
        title: 'Some idea',
        description: 'Some description',
        upvotes: 0,
        downvotes: 5,
        user: 'USER_ID'
      },
      {
        id: '1',
        title: 'Some another idea',
        description: 'Some another description',
        upvotes: 5,
        downvotes: 7,
        user: 'OTHER_USER_ID'
      }
    ]
  }
}
