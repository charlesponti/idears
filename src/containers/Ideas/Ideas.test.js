import React from 'react'
import { render } from '@testing-library/react'
import Ideas from '.'
import { GlobalContext } from 'src/context/GlobalState'
import { getMockState } from 'src/mocks'

describe('<Ideas/>', () => {
  it('should render ideas', () => {
    const state = getMockState()
    const { container } = render(
      <GlobalContext.Provider value={state}>
        <Ideas/>
      </GlobalContext.Provider>
    )
    expect(container).toHaveTextContent('Votes: 0')
    expect(container).toHaveTextContent('Votes: 5')
  })
})