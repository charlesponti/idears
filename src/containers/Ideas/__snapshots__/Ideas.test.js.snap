// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ideas/> should render ideas 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="container"
    >
      <div
        class="title"
      >
        <p>
          Some Crazy Idea
        </p>
        <p
          class="votes"
        >
          Votes: 
          0
        </p>
      </div>
      <div
        class="description"
      >
        An elaborate description of why this idea will change the world!
      </div>
      <div
        class="btnContainer"
      >
        <button
          aria-label="upvote idea"
          class="btnSuccess"
        >
           Upvote 
        </button>
        <button
          aria-label="downvote idea"
          class="btnDanger"
          disabled=""
        >
           Downvote 
        </button>
      </div>
    </div>
  </div>
</div>
`;
